scripts/tests-browser.ts

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable no-console */
+import { execSync } from 'child_process';
+import { readFileSync, readdirSync, writeFileSync } from 'fs';
+import { join } from 'path';
+
+const initialDependencies: { file: string; contents: string }[] = [];
+
+const packagesSupportingBrowserTesting = [
+  'abi-coder',
+  'address',
+  'crypto',
+  'errors',
+  'hasher',
+  'hdwallet',
+  'math',
+  'mnemonic',
+  'predicate',
+  'providers',
+  'signer',
+  'transactions',
+  'utils',
+];
+
+const hardlinkDeps = () => {
+  const packagesDirPath = join(__dirname, '../packages/');
+  const packages = readdirSync(packagesDirPath).filter((p) => !p.startsWith('.'));
+
+  packagesSupportingBrowserTesting.forEach((packageName) => {
+    const packageFilePath = join(packagesDirPath, `${packageName}/package.json`);
+    const fileContents = readFileSync(packageFilePath, 'utf8');
+    initialDependencies.push({ file: packageFilePath, contents: fileContents });
+
+    const lines = fileContents.split('\n');
+    lines.forEach((line, index) => {
+      if (line.includes('workspace:*')) {
+        packages.forEach((pckg) => {
+          if (line.includes(`@fuel-ts/${pckg}`)) {
+            lines[index] = line.replace('workspace:*', `file:../${pckg}`);
+          } else if (line.includes('fuels')) {
+            lines[index] = line.replace('workspace:*', `file:../fuels`);
+          }
+        });
+      }
+      writeFileSync(packageFilePath, lines.join('\n'));
+    });
+  });
+};
+
+const symlinkDeps = () =>
+  initialDependencies.forEach(({ file, contents }) => writeFileSync(file, contents));
+
+const executeBrowserTest = () =>
+  execSync(
+    'vitest --run --coverage --config vite.browser.config.mts $(scripts/tests-find.sh --browser)',
+    { stdio: 'inherit' }
+  );
+
+(() => {
+  try {
+    if (packagesSupportingBrowserTesting.length) {
+      hardlinkDeps();
```

Comment:
Why is this needed?

Can't `vite` work seamlessly with `pnpm` workspace?

Reply:
Cast your mind back to our early Vitest browser testing. We ran into the following error for certain tests: `Class extends value undefined is not a constructor or null`. Online didn't give much away, however I assumed it was to do with browser bundling due to the discrepancy with the `node` environment, and polyfill errors were _similar_. Hardlinking the dependencies resolved the problem for our `browser` tests.

However, I had not ran the suite with workspace linking since Vitest had released a major version of the browser testing package, and it seems to now be [working](https://github.com/FuelLabs/fuels-ts/pull/1630/commits/45b6a3e65e47efa2609fae4a91eca2a483b96aba). They had made a lot of changes to the browser package regarding both configuration and bundling in the major release, so potentially something in here resolved it. 🎉

Reply:
Oh yeah, I do remember that weird error!

I'm glad that doesn't happen anymore 🎉